civis_jupyter_notebooks/log_utils.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import os
+import sys
+
+USER_VISIBLE_LOGS = os.path.expanduser(os.path.join('~', 'work', 'civis-notebook-logs.log'))
+
+
+def setup_stream_logging():
+    """ Set up log format """
+    logger = logging.getLogger('CIVIS_PLATFORM_BACKEND')
+    logger.setLevel(logging.INFO)
+    # needed to remove duplicate log records..don't know why...
+    logger.propagate = False
```

Comment:
Maybe you don't need the `StreamHandler` below? See this note: https://docs.python.org/2/library/logging.html#logging.Logger.propagate

Reply:
Maybe you just need to set the handler on something like `logging` itself?

Reply:
Lemme talk to @beckermr about this code. I didn't write it, just moved it. But if I set the handler on logging itself I wouldn't get the ability to customize the logger, right? We want to set the format, log level, and name of the logger.

Reply:
This code was written through an extensive amount of guess and check. There is something about Jupyter and its use of messages (ZMQ I think...) and how they interact with logging. IIRC, there appeared to be a gap where logging would not happen or would happen twice as things were loaded and started with the sever. 

The set of incantations above and below fixed this and made the logging work. 

Sorry I don't have more details. It was a super weird bug at the heart of C python. Notice below I attached the logger the underlying stream from when python started (see here https://docs.python.org/3/library/sys.html#sys.__stdin__). Something is def weird.

Reply:
@mattyb I don't think this is appropriate PR to address this issue.

Reply:
Agreed

Reply:
Yep. I am not sure what there is to address. The code is ugly for sure but it works. Prob could use more comments. :(